batcher/aligned-batcher/src/types/user_state.rs

```diff
@@ -0,0 +1,29 @@
+use ethers::types::U256;
+
+pub(crate) struct UserState {
+    pub nonce: Option<U256>,
+    /// The minimum fee of a pending proof for a user.
+    /// This should always be the fee of the biggest pending nonce by the user.
+    /// This is used to check if a user is submitting a proof with a higher nonce and higher fee,
```

Comment:
If this is a minimum fee, wouldn't the invalid scenario be a lower fee?

Reply:
the wording in this part is a bit confusing, but the invalid scenario is when a user has submits a proof with some max fee N, and then submits a proof with a max fee M, with M > N.

Since the BatchPriorityQueue orders entries by max fee, the one with the higher max fee will be processed first, but this conflicts with the nonce ordering.
So, we restrict messages that have higher nonce to have a lower max fee than the one with a lower nonce